packages/typespec-ts/src/index.ts

```diff
@@ -199,7 +199,7 @@ export async function $onEmit(context: EmitContext) {
     options.generateTest =
       options.generateTest === true ||
       (options.generateTest === undefined &&
-        !hasTestFolder &&
+        (!hasTestFolder || (options.azureSdkForJs && options.azureArm)) &&
```

Comment:
1. generate test metadata for mgmt always
2. not generate sample.spec.ts file if test folder exists for mgmt